packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -11535,6 +11535,38 @@ describe('NetworkController', () => {
           },
         );
       });
+
+      it('emits the NetworkController:networkRemoved event', async () => {
+        const networkConfig = buildCustomNetworkConfiguration();
+        await withController(
+          {
+            state: {
+              selectedNetworkClientId: InfuraNetworkType.goerli,
+              networkConfigurationsByChainId: {
+                '0x1337': networkConfig,
+                [ChainId.goerli]: buildInfuraNetworkConfiguration(
+                  InfuraNetworkType.goerli,
+                ),
+              },
+            },
+          },
+          ({ controller, messenger }) => {
+            const networkRemovedListener = jest.fn();
+            messenger.subscribe(
+              'NetworkController:networkRemoved',
+              networkRemovedListener,
+            );
+
+            expect(
+              controller.state.networkConfigurationsByChainId,
+            ).toHaveProperty('0x1337');
+
+            messenger.call('NetworkController:removeNetwork', '0x1337');
```

Comment:
Do we want to call the `removeNetwork` method for consistency with other tests here? I also wonder whether we need to check for what's in state since we specified that state above and all we are testing is what the event listener is called with?

``` suggestion
            controller.removeNetwork('0x1337');
```

Reply:
Oh sorry I meant that we probably don't need to check for what's in state :) I notice we have these lines:

``` typescript
            expect(
              controller.state.networkConfigurationsByChainId,
            ).toHaveProperty('0x1337');
```

I'm guessing we probably don't need this since we control the state that the controller is being initialized with.

Reply:
Aha, gotcha. Yep sure I can remove this.

Reply:
OK removed that line, you were right, we don't need to assert the state before since was initialised with the initial state.